modules/features/discover/src/main/java/au/com/shiftyjelly/pocketcasts/discover/view/CategoriesListRowRedesignAdapter.kt

```diff
@@ -0,0 +1,151 @@
+package au.com.shiftyjelly.pocketcasts.discover.view
+
+import android.annotation.SuppressLint
+import android.content.Context
+import android.graphics.Color
+import android.view.LayoutInflater
+import android.view.ViewGroup
+import androidx.core.content.ContextCompat.getDrawable
+import androidx.core.content.ContextCompat.getString
+import androidx.core.view.isVisible
+import androidx.recyclerview.widget.DiffUtil
+import androidx.recyclerview.widget.ListAdapter
+import androidx.recyclerview.widget.RecyclerView
+import au.com.shiftyjelly.pocketcasts.discover.R
+import au.com.shiftyjelly.pocketcasts.discover.databinding.CategoryPillBinding
+import au.com.shiftyjelly.pocketcasts.servers.model.DiscoverCategory
+import au.com.shiftyjelly.pocketcasts.servers.model.NetworkLoadableList
+
+private val CATEGORY_REDESIGN_DIFF = object : DiffUtil.ItemCallback<CategoryPill>() {
+    override fun areItemsTheSame(oldItem: CategoryPill, newItem: CategoryPill): Boolean {
+        return oldItem == newItem
```

Comment:
This should compare by ID.

Reply:
I agree. [377f966](https://github.com/Automattic/pocket-casts-android/pull/1945/commits/377f966417f4a9444966a282e4ad9c9e42afa6f7)